packages/slate-react/src/components/android/android-editable.tsx

```diff
@@ -0,0 +1,449 @@
+import React, {
+  useCallback,
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+import {
+  Descendant,
+  Editor,
+  Element,
+  Node,
+  NodeEntry,
+  Range,
+  Text,
+  Transforms,
+} from 'slate'
+import throttle from 'lodash/throttle'
+import scrollIntoView from 'scroll-into-view-if-needed'
+
+import { ReactEditor } from '../..'
+import { ReadOnlyContext } from '../../hooks/use-read-only'
+import { useSlate } from '../../hooks/use-slate'
+import { useIsomorphicLayoutEffect } from '../../hooks/use-isomorphic-layout-effect'
+import {
+  DOMElement,
+  DOMNode,
+  DOMRange,
+  DOMStaticRange,
+  getDefaultView,
+  isDOMNode,
+  isPlainTextOnlyPaste,
+} from '../../utils/dom'
+import {
+  EDITOR_TO_ELEMENT,
+  EDITOR_TO_RESTORE_DOM,
+  EDITOR_TO_WINDOW,
+  ELEMENT_TO_NODE,
+  IS_FOCUSED,
+  IS_READ_ONLY,
+  NODE_TO_ELEMENT,
+  PLACEHOLDER_SYMBOL,
+} from '../../utils/weak-maps'
+import { AndroidInputManager } from './android-input-manager'
+import { EditableProps } from '../editable'
+import { ErrorBoundary } from './ErrorBoundary'
+import useChildren from '../../hooks/use-children'
+import {
+  defaultDecorate,
+  hasEditableTarget,
+  hasTarget,
+  isEventHandled,
+  isTargetInsideVoid,
+} from '../default-editable'
+
+const HAS_BEFORE_INPUT_SUPPORT = true
```

Comment:
This looks like it's not necessary in the Android specific editable.

Reply:
Removed dead code.